Add unit tests for shipping rate calculator

tarunvijwani opened the pull request:
In PR #8141, we're displaying a link to add the shipping address in the shipping calculator on the cart page when the shipping address is unavailable. We're also hiding the option to change the shipping address when default shipping rates are defined. 

In this PR, we're adding some more unit tests for #8141. 

Fixes #8027 and  #7730

## Changes in the PR

- Add unit test to check correct button is getting displayed. 


### Testing

#### Automated Tests
* [x] Changes in this PR are covered by Automated Tests.
  * [x] Unit tests
  * [ ] E2E tests

#### User Facing Testing

<!-- Write these steps from the perspective of a "user" (merchant) familiar with WooCommerce. No need to spell out the steps for common setup scenarios (eg. "Create a product"), but do be specific about the thing being tested. Include screenshots demonstrating expectations where that will be helpful. -->

1. Run 'npm run test -- assets/js/base/components/cart-checkout/totals/shipping/test/' in the terminal. 
2. Confirm all unit tests pass. 


* [x] Do not include in the Testing Notes <!-- Check this box if this PR can't be tested by users (ie: it doesn't include user-facing changes or it can't be tested without manually modifying the code). -->

### WooCommerce Visibility

<!-- Check this [this doc](../docs/blocks/feature-flags-and-experimental-interfaces.md) to see if the change is visible in WC core or not (part of the feature plugin or experimental)-->

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

Commit:
Add further unit tests

opr committed:
Add useStoreCart to mocked values and wrap in SlotFillProvider

opr committed:
Add test for 2nd and 3rd scenario

opr commented on assets/js/base/components/cart-checkout/totals/shipping/test/index.tsx:
```diff
@@ -0,0 +1,284 @@
+jest.mock( '@wordpress/data', () => ( {
+	__esModule: true,
+	...jest.requireActual( '@wordpress/data' ),
+	AsyncModeProvider:
```
I think this can be removed actually, since we are requiring `@wordpress/data` in `setup-globals.js`

tarunvijwani replied:
Thank you for the suggestion! I've removed it. 🙂

Commit:
Remove AsyncModeProvider mock

Remove AsyncModeProvider mock since we are requiring @wordpress/data in setup-globals.js

opr reviewed (approved):
Looks great @tarunvijwani thanks for addressing the changes!
